### ISSUE TITLE
fix(models): recurrent_gemma is registered but cannot generate text

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1687]
## Problem

`recurrent_gemma` (Griffin / RecurrentGemma) is registered as a supported family but cannot produce output: the model loads, then every generation either aborts in MLX or emits nothing but `<bos>`. Two defects stack, so fixing only the first converts a loud crash into a silent wrong answer. This issue covers completing the family end to end, not either defect alone.

Environment: mlxcel 0.7.0-beta.1 on `bench/0.7.0-refresh`, MacBook Pro M5 Max 128GB, macOS 26.6.2, MLX pin `9a795735`, built with `cargo build --release --features metal,accelerate`. Checkpoint `alpindale/recurrentgemma-9b` (17.96 GiB, 4 shards, `model_type: recurrent_gemma`, `architectures: ["RecurrentGemmaForCausalLM"]`, not quantized, `lru_width: 4096`, `conv1d_width: 4`, 38 layers, 16 Q heads over 1 KV head). `google/recurrentgemma-2b` and `-2b-it` are gated, so the ungated 9b mirror was used: `./target/release/mlxcel download alpindale/recurrentgemma-9b`, which lands in `~/.cache/mlxcel/models/alpindale/recurrentgemma-9b`.

## Current Behavior

Loading succeeds (`[Griffin] Model loaded successfully`, resident 5.01 GB, peak about 18 GB), then generation fails.

**Defect 1, the crash.** `src/models/recurrent_gemma.rs:376-378` transposes the padded input to `[B, C, L]` before `mlxcel_core::conv1d`, with the comment "Depthwise conv1d: transpose to [B, C, L], apply conv, transpose back". That is the PyTorch NCL layout. MLX `conv1d` takes NLC, `[B, L, C]`, and reads the channel count off the last axis, so the group count never matches: `libc++abi: terminating due to uncaught exception of type std::invalid_argument: [conv] The input channels must be divisible by the number of groups. Got input with shape (1,4096,11) and 4096 groups.` The third dimension tracks prompt length (a 5-token prompt gives `(1,4096,5)`, a 21-token prompt `(1,4096,21)`), confirming the tensor is `[batch, channels, length]` while `groups` is `lru_width` 4096. Two in-tree call sites are already correct and are the reference: `src/models/mamba.rs:252-260` and `src/models/nemotron_h.rs:581-589` both pass `padded_input` straight to `conv1d` with no transpose, and `src/models/conv_decode_tests.rs:70-76` builds its reference call with input `[1, k, ch]` against weight `[ch, k, 1]`.

**Defect 2, wrong output.** With both transposes removed the model runs but emits `<bos>` for every position. Reproduced at `--temp 0`, with and without `--no-chat-template`, on a chat prompt ("List the first five prime numbers.") and a completion prompt ("The capital of France is"), the latter yielding 20 consecutive `<bos>` tokens. At least one more defect exists in the Griffin conv / RGLRU path beyond the conv layout. It has not been localized, so localizing it is part of this issue.

## Supporting Context

The family is wired into `src/models/registry.rs:364,499,734` (`ModelType::RecurrentGemma`, aliases `recurrent_gemma` and `griffin`, family `hybrid_ssm`), mapped at `src/models/detection.rs:730`, and advertised at `src/models/mod.rs:913`. Its 3 in-file unit tests (`recurrent_gemma.rs:1108-1180`) cover EOS metadata and sequence-state cache plumbing; none exercise the conv or RGLRU numerics. In `docs/supported-models.md` it appears only as a bare name in the one-line list at line 58, with no detail entry and no validated checkpoint, unlike every neighbouring family. That is consistent with a port never run against a real checkpoint. #1220 fixed recurrent-state rebuild here, so the cache plumbing is newer than the forward math.

## Blocked Follow-up

`LocalAttentionBlock` still calls `repeat_kv` on GQA-shaped K and V immediately before the fused SDPA (`src/models/recurrent_gemma.rs:494-501`, feeding `attention_from_ptr` at `:505-509`), the redundant pattern removed from six VL decoders in commit `5946fa58` under #1686 for 1.4x to 1.8x at 2048 prompt tokens. #1686 lists `recurrent_gemma` as an out-of-scope caller to file separately, and this checkpoint has the largest expansion of any remaining caller (`n_rep` 16, from 16 Q heads over 1 KV head). It cannot be removed until output is correct, because there would be no way to validate the change.

## Scope

**In scope:** the conv1d layout fix in `src/models/recurrent_gemma.rs`, localizing and fixing whatever else breaks Griffin generation, a numerics-level test beside the existing three, the `docs/supported-models.md` entry, and the `repeat_kv` removal once output is correct.

**Out of scope:** the other remaining `repeat_kv` callers (`deepseek_v2`, `minicpm3`, `nemotron_nas`), which need their own measurement per #1686.

## Implementation Notes

- **Reuse:** match the `mamba.rs` / `nemotron_h.rs` conv shape exactly rather than writing a third variant; both pass `padded_input` unmodified and reshape the bias to `[1, 1, -1]`, which `recurrent_gemma.rs:380-381` already does.
- **Do not ship the conv fix alone.** It removes the abort without producing correct text, and a family that silently emits `<bos>` is worse than one that crashes.
- **Localize defect 2** against `mlx_lm/models/recurrent_gemma.py` by comparing per-layer activations, not just final logits. The crash masked everything downstream, so candidates are RGLRU gate and decay arithmetic, the conv cache tail slice at `recurrent_gemma.rs:383-387`, the `[B, L, C]` assumption in `RGLRU::forward`, and the local-attention window against `attention_window_size` 2048.
- **Edge cases:** single-token decode (`L == 1`) versus multi-token prefill through the same conv path, the first step where the conv cache is absent and zeros are padded, and the recurrent-state handoff across the 2 recurrent to 1 attention pattern this config reports.

## Acceptance Criteria

- [ ] `mlxcel_core::conv1d` in `recurrent_gemma.rs` is called with MLX NLC input matching the shape `mamba.rs` and `nemotron_h.rs` pass, and neither the pre-transpose nor the post-transpose remains.
- [ ] Greedy generation on `alpindale/recurrentgemma-9b` produces coherent text for both a chat-template prompt and a `--no-chat-template` completion prompt, with no `<bos>` repetition.
- [ ] Output is checked against the `mlx_lm/models/recurrent_gemma.py` reference on the same checkpoint and prompt, not merely checked for being non-empty, and the comparison is recorded in the PR body.
- [ ] A unit test covers the conv path at the shape level so a reintroduced transpose fails in `cargo test` rather than at first inference.
- [ ] `docs/supported-models.md` gains a real entry for the family naming the validated checkpoint, replacing the bare mention at line 58.
- [ ] `repeat_kv` no longer appears in `recurrent_gemma.rs`, the removal is measured at 64 and 2048 prompt tokens, and the `// Used by:` list on `repeat_kv` in `src/lib/mlxcel-core/src/utils.rs` is regenerated per `docs/code-guidelines.md`.

## Verification

```bash
cargo build --release --features metal,accelerate
cargo test --release recurrent_gemma
cargo clippy --all-targets -- -D warnings && cargo fmt --check

M=~/.cache/mlxcel/models/alpindale/recurrentgemma-9b
./target/release/mlxcel generate -m "$M" -p "The capital of France is" -n 20 --temp 0 --no-chat-template
./target/release/mlxcel generate -m "$M" -p "List the first five prime numbers." -n 60 --temp 0
```

A pass looks like: no MLX exception, the completion prompt continuing with Paris rather than `<bos>`, the chat prompt listing 2, 3, 5, 7, 11, and the greedy continuation agreeing with the mlx-lm reference on the same checkpoint. For the `repeat_kv` step, report decode tok/s at 64 and 2048 prompt tokens before and after, per `docs/benchmarks.md`.

**Comment by inureyes** [comment]
The mlx-lm reference is present on the validation host (`~/.cache/uv/archive-v0/*/mlx_lm/models/recurrent_gemma.py`), so the comparison in Verification is runnable without extra setup. Two findings from reading it.

**Defect 1 confirmed from the reference side.** Its `Conv1d.__call__` takes `x` as `[B, L, C]`, pads or concatenates the cache on axis 1, and calls `mx.conv_general(x, self.weight, groups=groups)` with no transpose in either direction, against a weight of `(channels, kernel_size, 1)`. That is the same NLC shape `mamba.rs` and `nemotron_h.rs` already pass, so removing both transposes is the correct fix rather than a shape workaround.

**One candidate for defect 2 is ruled out.** The conv weight layout is already handled: the checkpoint ships `model.layers.N.temporal_block.conv_1d.weight` as `[4096, 1, 4]`, and `sanitize_weights` at `src/models/recurrent_gemma.rs:754-767` applies `swap_axes(v, -1, -2)` to reach `[4096, 4, 1]`, matching the reference `sanitize`'s `v.moveaxis(2, 1)` and the `[ch, k, 1]` weight in `conv_decode_tests.rs`. Look elsewhere for the `<bos>` defect.

**Reference contract worth diffing against `RGLRU::forward`,** since these fail silently rather than throwing: gates come from a per-head block-diagonal linear with `[num_heads, head_dim, head_dim]` weights (reshape to heads, `swapaxes(1, 2) @ w`, add bias, sigmoid, flatten), the decay is `log_a = -8.0 * gate_a * softplus(recurrent_param)`, and the gamma normalizer is `sqrt(1 - exp(2 * log_a))` with position 0 forced to `1.0` only when there is no cache.
